root/usr/lib64/ruby/vendor_ruby/discovery.rb

```diff
@@ -166,10 +166,10 @@ def upload(uri = discover_server, type = proxy_type, custom_facts = {})
     return
   end
   if uri.host.nil? or uri.port.nil?
-    log_err "Foreman URI host or port was not specified, cannot continue"
+    log_err "Server or proxy URI host or port was not specified, cannot continue"
```

Comment:
I think `proxy` is also problematic but not sure you can use anything else here.